src/lib/driver/message_handler.ts

```diff
@@ -65,6 +65,15 @@ export class DriverMessageHandler {
       case DriverCommand.enableErrorReporting:
         driver.enableErrorReporting();
         return {};
+      case DriverCommand.softReset:
+        await driver.softReset();
```

Comment:
These commands should raise an error if we are currently doing firmware upgrades, to prevent us from bricking devices.

Reply:
this is a great call 👍🏾

Reply:
Are you talking about end devices or the controller?  For end devices the upgrade process is resilient enough to handle being interrupted. as for the controller, shouldn't that be handled by the underlying zwave-js driver?  If not I will have to defer to someone smarter than me to implement these changes and I will close out this PR for now.

Reply:
the controller. We'd have to look at the underlying logic for `softReset` and `hardReset` to see if it protects against that, I don't know offhand

Reply:
I don't see anything that indicates protection around this.